### PR TITLE
Adding an example for HttpHandler extension.

### PR DESCRIPTION
**Pull request by buzztroll**

### CODE REVIEW

File: examples/handler/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+To build the example simply run: `./gradlew jar`.  This will create the file:
+`./build/libs/handler-1.0.jar` which contains the handler extension and the
+`META-INF/services` definition that tells Stardog how to load this into the
+ handler chain.
+ 
+## Load into Stardog
+ 
+To load the module into Stardog set the environment variable `STARDOG_EXT`
+to the path that contains the jar referenced above.  Then use `stardog-admin`
+it start the server in the typical manner.
```
Comment (jbalint) [inline examples/handler/README.md]: *to* start the server

File: examples/handler/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+# Example Stardog Undertow HTTP Handler Extension
```
Comment (jbalint) [inline examples/handler/README.md]: should the examples dir be called http handler?

File: examples/handler/build.gradle [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+	}
+	maven {
+		url "http://maven.stardog.com"
+	}
+
+
+}
+
+dependencies {
+	compile "com.complexible.stardog:client-http:5.2.0"
```
Comment (jbalint) [inline examples/handler/build.gradle]: use `${stardogVersion}` variable (and update to 5.2.0 if you need it)

File: examples/handler/build.gradle [hunk 4]
```diff
@@ -0,0 +1,32 @@
+group = "com.complexible.stardog.examples.handler"
+version '1.0'
+
+apply plugin: 'java'
+
+sourceCompatibility = 1.8
+
+repositories {
```
Comment (jbalint) [inline examples/handler/build.gradle]: why do we need these repos?

File: examples/handler/gradlew [hunk 5]
```diff
@@ -0,0 +1,164 @@
+#!/usr/bin/env bash
```
Comment (jbalint) [inline examples/handler/gradlew]: I don't think we should add gradlew here

File: examples/handler/src/main/java/com/complexible/stardog/examples/handler/TestHttpHandler.java [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+	private final HttpServiceLoader.ServerContext mContext;
+
+
+	public TestHttpHandler(final HttpHandler theNext, final HttpServiceLoader.ServerContext theContext) {
+		LOGGER.debug("Loading TestHttpHandler");
+		mNext = theNext;
+		mContext = theContext;
+	}
+
+	public void handleRequest(HttpServerExchange theExchange) throws Exception {
```
Comment (jbalint) [inline examples/handler/src/main/java/com/complexible/stardog/examples/handler/TestHttpHandler.java]: maybe some brief details about what the exchange is, how to handle it, whether `handleRequest()` needs called, etc

**Comment by buzztroll** [comment]
This is rebased and ready now.

File: build.gradle [hunk 7]
```diff
@@ -7,6 +7,8 @@ task wrapper(type: Wrapper) {
 ext {
 	stardogVersion = project.hasProperty("stardogVersion") ? project.getProperty("stardogVersion") : '5.2.4'
 	projectVersion = '0.0.4'
+	stardogVersion = project.hasProperty("stardogVersion") ? project.getProperty("stardogVersion") : '5.2.1-SNAPSHOT-20180118'
```
Comment (mhgrove) [inline build.gradle]: shouldn't this be 5.3?

File: examples/http_handler/README.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+
+To build the example simply run: `./gradlew jar`.  This will create the file:
+`./build/libs/handler-1.0.jar` which contains the handler extension and the
+`META-INF/services` definition that tells Stardog how to load this into the
+ handler chain.
+
+## Load into Stardog
+
+To load the module into Stardog set the environment variable `STARDOG_EXT`
+to the path that contains the jar referenced above.  Then use `stardog-admin`
```
Comment (mhgrove) [inline examples/http_handler/README.md]: isn't the `ext` dir in the distro the default for this?

**Comment by buzztroll** [comment]
Concerns addressed

**Pull request merged**